plugins/cassandra/cassandrawu.cpp

```diff
@@ -2470,6 +2304,169 @@ class CCassandraWorkUnit : public CPersistedWorkUnit
         }
     }
 
+    virtual void clearGraphProgress() const
+    {
+        const char *wuid = queryWuid();
+        CassandraBatch batch(cass_batch_new(CASS_BATCH_TYPE_UNLOGGED));
+        sessionCache->deleteChildByWuid(wuGraphProgressMappings, wuid, batch);
+        sessionCache->deleteChildByWuid(wuGraphStateMappings, wuid, batch);
+        sessionCache->deleteChildByWuid(wuGraphRunningMappings, wuid, batch);
+        CassandraFuture futureBatch(cass_session_execute_batch(sessionCache->querySession(), batch));
+        futureBatch.wait("clearGraphProgress");
+    }
+    virtual bool getRunningGraph(IStringVal &graphName, WUGraphIDType &subId) const
+    {
+        CassandraStatement statement(sessionCache->prepareStatement("SELECT graphID, subgraphID FROM wuGraphRunning where partition=? and wuid=?;"));
+        const char *wuid = queryWuid();
```

Comment:
Maybe, though IMHO the code is more readable and easier to follow (and easier to see that all params have been bound) if I spell them out.